__tests__/posts.ts

```diff
@@ -3473,6 +3471,94 @@ describe('query sourcePostModeration', () => {
   });
 });
 
+describe('mutation createSourcePostModeration', () => {
+  beforeEach(async () => {
+    await saveSquadFixtures();
+  });
+
+  const MUTATION = `mutation CreateSourcePostModeration($sourceId: ID! $title: String!, $type: String!, $content: String, $image: Upload, $imageUrl: String, $sharedPostId: ID, $commentary: String, $externalLink: String) {
+    createSourcePostModeration(sourceId: $sourceId, title: $title, type: $type, content: $content, image: $image, imageUrl: $imageUrl, sharedPostId: $sharedPostId, commentary: $commentary, externalLink: $externalLink) {
+      id
+      title
+      content
+      contentHtml
+      externalLink
+      type
+      image
+      sharedPostId
+      titleHtml
+    }
+  }`;
+
+  const params = {
+    sourceId: 'm',
+    title: 'My first post',
+    content: 'Hello World',
+  };
+
+  it('should result in error because user is not member of squad', async () => {
+    loggedUser = '2';
+
+    return await testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: { ...params, type: PostType.Freeform },
+      },
+      'FORBIDDEN',
+    );
+  });
+
+  it('should successfully create a squad post moderation entry of type freeform', async () => {
+    loggedUser = '4';
+    const res = await client.mutate(MUTATION, {
+      variables: { ...params, type: PostType.Freeform },
+    });
+    expect(res.errors).toBeFalsy();
+    expect(res.data.createSourcePostModeration).toBeTruthy();
+    expect(res.data.createSourcePostModeration.type).toEqual(PostType.Freeform);
+    expect(res.data.createSourcePostModeration.contentHtml).toBeDefined();
+  });
+
+  it('should successfully create a squad post moderation entry of type share', async () => {
+    loggedUser = '4';
+    const res = await client.mutate(MUTATION, {
+      variables: { ...params, sharedPostId: 'p1', type: PostType.Share },
+    });
+    expect(res.errors).toBeFalsy();
+    expect(res.data.createSourcePostModeration).toBeTruthy();
+    expect(res.data.createSourcePostModeration.type).toEqual(PostType.Share);
+    expect(res.data.createSourcePostModeration.contentHtml).toBeDefined();
+    expect(res.data.createSourcePostModeration.sharedPostId).toEqual('p1');
+  });
+
+  it('should successfully create a squad post moderation entry of type article', async () => {
```

Comment:
It should not actually be possible to create a type of `article` as that only comes from the scraper.

Reply:
@sshanzel Are you sure? the RFC mentions the type can be article, share or freeform. I assumed Article was a shared link. Looking at createExternalLink , it adds a post of type article:

```
 await entityManager.getRepository(ArticlePost).insert({
```

Reply:
Yes. We may create the article post behind the scenes but we don't tag it as created by the user. We will create another post with `Share` type that is then referenced to the logged user.

Reply:
This means, your request is of `share` type than `article.